cddl/usr.sbin/zfsd/case_file.cc

```diff
@@ -1158,43 +1202,54 @@ CaseFile::Replace(const char* vdev_type, const char* path, bool isspare) {
 	return (retval);
 }
 
-/* Does the argument event refer to a checksum error? */
-static bool
-IsChecksumEvent(const Event* const event)
+/* Lookup the vdev prop. Used for for checksum, IO, or slow IO props */
+int
+CaseFile::getVdevProp(vdev_prop_t vdev_prop) const
 {
-	return ("ereport.fs.zfs.checksum" == event->Value("type"));
-}
+	char val[ZFS_MAXPROPLEN];
+	zprop_source_t srctype;
+	DevdCtl::Guid poolGUID = PoolGUID();
+	ZpoolList zpl(ZpoolList::ZpoolByGUID, &poolGUID);
+	zpool_handle_t *zhp(zpl.empty() ? NULL : zpl.front());
 
-/* Does the argument event refer to an IO error? */
-static bool
-IsIOEvent(const Event* const event)
-{
-	return ("ereport.fs.zfs.io" == event->Value("type"));
-}
+	char *prop_str = (char *) vdev_prop_to_name(vdev_prop);
+	if (zhp == NULL || zpool_get_vdev_prop(zhp, m_vdevName.c_str(),
+	    vdev_prop, prop_str, val, sizeof (val), &srctype, B_FALSE) != 0)
+		return (0);
 
-/* Does the argument event refer to an IO delay? */
-static bool
-IsDelayEvent(const Event* const event)
-{
-	return ("ereport.fs.zfs.delay" == event->Value("type"));
+	/* we'll get "-" from libzfs for a prop that is not set */
+	if (strcmp(val, "-") == 0)
+		return (0);
+
+	return (atoi(val));
 }
 
 bool
 CaseFile::ShouldDegrade() const
 {
+	int io_n = getVdevProp(VDEV_PROP_IO_N);
```

Comment:
I think this should be
```suggestion
	int io_n = getVdevProp(VDEV_PROP_CHECKSUM_N);
```